x-pack/plugins/cases/public/components/user_actions/use_user_permissions.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useMemo, useCallback } from 'react';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import type { UserActivityParams } from '../user_actions_activity_bar/types';
+
+export const useUserPermissions = () => {
```

Comment:
This began way more complicated than it is currently. Happy to de-construct it and just use `permissions` directly in the code, but fine with whatever makes sense